zokrates_core/src/ir/expression.rs

```diff
@@ -184,23 +210,27 @@ mod tests {
             let a: LinComb<FieldPrime> = FlatVariable::new(42).into();
             let b: LinComb<FieldPrime> = FlatVariable::new(42).into();
             let c = a + b.clone();
-            let mut expected_map = BTreeMap::new();
-            expected_map.insert(FlatVariable::new(42), FieldPrime::from(2));
-            assert_eq!(c, LinComb(expected_map));
+
+            let expected_vec = vec![
+                (FlatVariable::new(42), FieldPrime::from(1)),
+                (FlatVariable::new(42), FieldPrime::from(2)),
```

Comment:
Shouldn't this be `[(_42, 1), (_42, 1)]` as the unreduced original `(_42 => 2)`

Reply:
Yes it should, good catch!